src/remark/code-block-snippets.js

```diff
@@ -24,8 +24,16 @@ let mdFilePath = "";
  * @typedef {import('mdast').Code} CodeNode
  * @param {Readonly<Omit<ImageNode, 'type'>> & CodeNode} node
  */
+
+/**
+ * Find paragraph -> image pattern and perform the replacement.
+ */
 function visitor(node) {
-    if (!node.alt?.startsWith("code")) return;
+    if (!node.children || node.children[0]?.type != 'image' || !node.children[0]?.alt?.startsWith("code")) return;
+
+    // Move code block in place of the paragraph node.
+    Object.assign(node, node.children[0]);
```

Comment:
I don't think this assignment is needed. Image node has properties:
```ts
type: 'image';
url: string;
alt?: string;
title?: string;
data?: TData;
position?: Position;
```

and below you set:
```ts
type: 'code';
value: string;
url: string;
lang?: string;
meta?: string;
```

so only `alt, title, data, position` would remain, but don't seem to change anything.

So a more readable alternative would be
```ts
    const imgNode = node.children[0];
    delete node.children;
    node.type = 'code';
    node.lang = imgNode.alt.split(" ")[1];
    ...
```